File: app/src/main/java/test/itexico/movies/managers/RequestManager.java
package test.itexico.movies.managers;

import android.content.Context;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

public class RequestManager {
    private static RequestManager instance;

    private RequestQueue mRequestQueue;

    private RequestManager(Context context){
        if (mRequestQueue == null) {
            mRequestQueue = Volley.newRequestQueue(context.getApplicationContext());
        }
    }

    public static RequestManager getInstance(Context context) {
        if(instance==null){
            instance = new RequestManager(context);
        }
        return instance;

    }

    public RequestQueue getRequestQueue() {
        return mRequestQueue;
    }

    public <T> void addToRequestQueue(Request<T> req) {
        getRequestQueue().add(req);
    }
}

File: app/src/main/java/test/itexico/movies/managers/RequestManager.kt
package test.itexico.movies.managers

import android.content.Context

import com.android.volley.Request
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley

class RequestManager private constructor(context: Context) {

    var requestQueue: RequestQueue? = null
        private set

    init {
        if (requestQueue == null) {
            requestQueue = Volley.newRequestQueue(context.applicationContext)
        }
    }

    fun <T> addToRequestQueue(req: Request<T>) {
        requestQueue?.add(req)
    }

    companion object {
        private var instance: RequestManager? = null

        fun getInstance(context: Context): RequestManager {
            if (instance == null) {
                instance = RequestManager(context)
            }
            return instance as RequestManager

        }
    }
}
